common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -913,6 +1028,50 @@ public UTF8String subStringIndex(UTF8String delim, int count) {
     }
   }
 
+  public UTF8String collatedSubStringIndex(UTF8String delim, int count, int collationId) {
```

Comment:
Let's go with this naming: `collationAwareSubStringIndex`

ref: https://github.com/apache/spark/pull/45749#discussion_r1546421174